cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -132,4 +140,41 @@ void publishToControlChannel(TetheringControlMessage message) throws Exception {
       throw new Exception(String.format("Failed to publish to topic %s", topicId), e);
     }
   }
+
+  @VisibleForTesting
+  String getLocalFileAsString(LocalFile file) throws IOException {
+    return new String(Files.readAllBytes(new File(file.getURI()).toPath()), StandardCharsets.UTF_8);
+  }
+
+  /**
+   * Add logback.xml, program options, and select cConf fields from local files to the control message payload
+   */
+  private TetheringLaunchMessage createLaunchPayload(Collection<? extends LocalFile> localFiles) throws IOException {
+    Map<String, String> localizeFiles = new HashMap<>();
+    Map<String, String> cConfEntries = new HashMap<>();
+
+    LocalFile logbackFile = localFiles.stream()
+      .filter(file -> file.getName().equals(DistributedProgramRunner.LOGBACK_FILE_NAME))
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException("Cannot find logback file"));
+    localizeFiles.put(DistributedProgramRunner.LOGBACK_FILE_NAME, getLocalFileAsString(logbackFile));
+
+    LocalFile programOptionsFile = localFiles.stream()
+      .filter(file -> file.getName().equals(DistributedProgramRunner.PROGRAM_OPTIONS_FILE_NAME))
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException("Cannot find program options file"));
+    localizeFiles.put(DistributedProgramRunner.PROGRAM_OPTIONS_FILE_NAME, getLocalFileAsString(programOptionsFile));
+
+    LocalFile appSpecFile = localFiles.stream()
+      .filter(file -> file.getName().equals(DistributedProgramRunner.APP_SPEC_FILE_NAME))
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException("Cannot find app spec file"));
+    localizeFiles.put(DistributedProgramRunner.APP_SPEC_FILE_NAME, getLocalFileAsString(appSpecFile));
+
+    for (String prefix : SELECT_CCONF_FIELDS) {
+      String prefixRegex = "^" + prefix.replace(".", "\\.");
+      cConfEntries.putAll(cConf.getValByRegex(prefixRegex));
+    }
+    return new TetheringLaunchMessage("v1", localizeFiles, cConfEntries);
```

Comment:
we should have v1 directory which will have V1 Message class so that when we change it to v2, we know what was v1.

Reply:
done